### ISSUE TITLE
Debian/Ubuntu stable package is broken

### ISSUE THREAD

**Issue by Habetdin** [netdata/netdata#7067]
#### Bug report summary

After upgrading my netdata from 1.17 to 1.18 today, I've got the same error as the #6925 when trying to open netdata in my browser:
`ERROR 101: Cannot load jQuery: https://netdata.hostname/lib/jquery-2.2.4.min.js`

##### OS / Environment

Ubuntu 18.04 LTS, using stable packagecloud branch of netdata.

##### Netdata version (ouput of `netdata -V`)

`netdata v1.18.0`

##### Component Name

netdata/packaging/deb

##### Steps To Reproduce

Upgrade netdata from 1.17 to 1.18 using stable packagecloud branch.

##### Expected behavior

Web UI should keep its working state.

##### Temporary solution

Got www directory path from full config:
```
[global]
	# I'm not overriding the default value, so can't tell if did it change since the 1.17?
	# web files directory = /var/lib/netdata/www
```
It had some stuff in the root directory (`/var/lib/netdata/www`) but not in its subdirectories like `images` or `lib`.

At first I was about to just copy the files like in #6925 by doing `cp -a /usr/share/netdata/www/* /var/lib/netdata/www/`, but after checking the #6930 I've decided to remove empty directories:
```
rm -rf /var/lib/netdata/www/.well-known
rm -rf /var/lib/netdata/www/css
rm -rf /var/lib/netdata/www/fonts
rm -rf /var/lib/netdata/www/images
rm -rf /var/lib/netdata/www/lib
rm -rf /var/lib/netdata/www/static
```
To replace them with the next proper symlinks:
```
ln -s /usr/share/netdata/www/.well-known /var/lib/netdata/www/.well-known
ln -s /usr/share/netdata/www/css         /var/lib/netdata/www/css
ln -s /usr/share/netdata/www/fonts       /var/lib/netdata/www/fonts
ln -s /usr/share/netdata/www/images      /var/lib/netdata/www/images
ln -s /usr/share/netdata/www/lib         /var/lib/netdata/www/lib
ln -s /usr/share/netdata/www/static      /var/lib/netdata/www/static
```

That resolved the issue for me.

P.S.: Same actions in one-liner by @borisdigital https://github.com/netdata/netdata/issues/6925#issuecomment-541026780:
`for dn in ".well-known" css fonts images lib static; do rmdir /var/lib/netdata/www/${dn}; ln -s /usr/share/netdata/www/${dn} /var/lib/netdata/www/${dn}; done`

P.P.S.: Posting this issue because it seems like the closed state of #6925 does mute notifications for developers.

**Comment by dam09fr** [comment]
I got exactly the same issue on Debian 10.
Waiting for a fix, let me know if you need some info.

**Comment by paulkatsoulakis** [comment]
Hi guys, thank you for reporting this.
I 'll check it out and get back to you

**Comment by paulkatsoulakis** [comment]
That is interesting. It seems that the upgrade path somehow leaves inconsistencies back.
I tried two scenarios:
1) On a clean container i installed 1.17.1 and then upgraded to 1.18.0 and the error popped up (you need to clean browser cache though, because if you have cached data you may not get it)
2) On a clean container, i installed directly 1.18.0 and the problem did not occur

```
NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.2 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

I 'll be working on how to mitigate the upgrade path issue, any thoughts welcome.
In the meantime,  @dam09fr and @Habetdin do you think you can try uninstalling and installing netdata again and confirm my findings?  Removing the netdata package and adding back directly the 1.18.0 should not raise the problem.

Thanks again for your report and sorry for bumping into this.

**Comment by borisdigital** [comment]
> That is interesting. It seems that the upgrade path somehow leaves inconsistencies back.
> I tried two scenarios:
> 
>     1. On a clean container i installed 1.17.1 and then upgraded to 1.18.0 and the error popped up (you need to clean browser cache though, because if you have cached data you may not get it)
> 
>     2. On a clean container, i installed directly 1.18.0 and the problem did not occur
> 
> 
> ```
> NAME="Ubuntu"
> VERSION="18.04.2 LTS (Bionic Beaver)"
> ID=ubuntu
> ID_LIKE=debian
> PRETTY_NAME="Ubuntu 18.04.2 LTS"
> VERSION_ID="18.04"
> HOME_URL="https://www.ubuntu.com/"
> SUPPORT_URL="https://help.ubuntu.com/"
> BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
> PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
> VERSION_CODENAME=bionic
> UBUNTU_CODENAME=bionic
> ```
> 
> I 'll be working on how to mitigate the upgrade path issue, any thoughts welcome.
> In the meantime, @dam09fr and @Habetdin do you think you can try uninstalling and installing netdata again and confirm my findings? Removing the netdata package and adding back directly the 1.18.0 should not raise the problem.
> 
> Thanks again for your report and sorry for bumping into this.
@paulkatsoulakis 
I had the same problem on debian buster.

As requested, I removed netdata, deleted the leftover files at /etc/netdata and /var/{cache,lib}/netdata, and did a "clean" install. This seems to work fine here.

**Comment by paulkatsoulakis** [comment]
cool, that means we have spotted the root cause, thank you for cross checking this @borisdigital, you rock.

We 'll let you know as soon as the fix is out

**Comment by dam09fr** [comment]
Confirmed on my side (Debian 10) :
- Fresh install of 1.18 is working
- Upgrade from 1.17.1 to 1.18 does not work

On fresh 1.18, the following symlinks are present:
```
# ls -l /var/lib/netdata/www | grep ^l
lrwxrwxrwx 1 root netdata     26 Oct 10 13:48 css -> /usr/share/netdata/www/css
lrwxrwxrwx 1 root netdata     28 Oct 10 13:48 fonts -> /usr/share/netdata/www/fonts
lrwxrwxrwx 1 root netdata     29 Oct 10 13:48 images -> /usr/share/netdata/www/images
lrwxrwxrwx 1 root netdata     26 Oct 10 13:48 lib -> /usr/share/netdata/www/lib
lrwxrwxrwx 1 root netdata     29 Oct 10 13:48 static -> /usr/share/netdata/www/static
```

None of them are present after upgrading from 1.17.1 to 1.18.

**Comment by Habetdin** [comment]
@paulkatsoulakis well, fresh install works fine, so I'm sure it's more like "upgrading to v1.18" issue.

**Comment by Habetdin** [comment]
Fresh install of netdata 1.17 @ Ubuntu 18.04 LTS:
```shell
root@test:~# netdata -V
netdata v1.17.1
root@test:~# ls -ld /var/lib/netdata/www/*/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:10 /var/lib/netdata/www/css/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:10 /var/lib/netdata/www/fonts/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:10 /var/lib/netdata/www/images/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:10 /var/lib/netdata/www/lib/
drwxr-xr-x 3 root netdata 4096 Oct 13 00:10 /var/lib/netdata/www/static/
root@test:~# ls -ld /usr/share/netdata/www/*/
ls: cannot access '/usr/share/netdata/www/*/': No such file or directory
```

After upgrading it to the 1.18:
```shell
root@test:~# netdata -V
netdata v1.18.0
root@test:~# ls -ld /var/lib/netdata/www/*/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /var/lib/netdata/www/css/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /var/lib/netdata/www/fonts/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /var/lib/netdata/www/images/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /var/lib/netdata/www/lib/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /var/lib/netdata/www/static/
root@test:~# ls -ld /usr/share/netdata/www/*/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /usr/share/netdata/www/css/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /usr/share/netdata/www/fonts/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /usr/share/netdata/www/images/
drwxr-xr-x 2 root netdata 4096 Oct 13 00:14 /usr/share/netdata/www/lib/
drwxr-xr-x 3 root netdata 4096 Oct 13 00:14 /usr/share/netdata/www/static/
```

After the upgrade, directories become empty but were not replaced with the symlinks.
```shell
root@test:~# ls -l /var/lib/netdata/www/*/
/var/lib/netdata/www/css/:
total 0

/var/lib/netdata/www/fonts/:
total 0

/var/lib/netdata/www/images/:
total 0

/var/lib/netdata/www/lib/:
total 0

/var/lib/netdata/www/static/:
total 0
```

**Comment by knatsakis** [comment]
Hi,

I believe this issue is caused because the upgrade from v1.17.1 to v1.18.0 tries to change /var/www/lib/netdata/www/{css,fonts,images,....} from directories to symlinks.

As mentioned in [man dpkg-maintscript-helper](http://manpages.ubuntu.com/manpages/bionic/man1/dpkg-maintscript-helper.1.html)

> When upgrading a package, dpkg will not automatically switch a symlink to a directory or vice-versa. Downgrades are not supported and the path will be left as is.

I will try to come up with a fix for this.

**Comment by knatsakis** [comment]
This issue should be fixed by release v1.18.1.

The upgrade from v1.17.1 or v1.18.0, to v1.18.1+ should be working without issues.

@Habetdin , @dam09fr, @borisdigital thank you very much the report and your help!
